go/signedexchange/certurl/parse.go

```diff
@@ -0,0 +1,77 @@
+package certurl
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+)
+
+func writeHead(w *bytes.Buffer, n int, size int) error {
+	for i := 0; i < size; i++ {
+		if err := w.WriteByte(byte(n >> (8 * uint(size-i-1)))); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ParsePEM parses a PEM formatted content to a certUrl content.
```

Comment:
I'd expect a function called `ParsePEM` to only parse the PEM format, and not also serialize it to the TLS 1.3 Certificate message format. Call it `CertificateMessageFromPEM` or similar, or split it into two functions?